src/egg_hatch.c

```diff
@@ -388,13 +388,13 @@ void ScriptHatchMon(void)
     AddHatchedMonToParty(gSpecialVar_0x8004);
 }
 
-static bool8 sub_807158C(struct DayCare *daycare, u8 daycareId)
+static bool8 _CheckDaycareMonReceivedMail(struct DayCare *daycare, u8 daycareId)
 {
-    u8 nick[0x20];
+    u8 nick[32];
     struct DaycareMon *daycareMon = &daycare->mons[daycareId];
 
     GetBoxMonNick(&daycareMon->mon, nick);
```

Comment:
This should really be expanded to `GetBoxMonNickname`

Reply:
Expanded the uses of nick in `egg_hatch.c` and `daycare.c`. `GetMonNickname` was taken by an essentially identical function in `party_menu.c`, so the less-used one was given `GetMonNickname2`.